dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -297,7 +325,16 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
     } yield updated
   }
 
-  def signDigits(
+  override def signDigits(eventName: String, num: Long): Future[OracleEvent] = {
+    for {
+      eventOpt <- findEvent(eventName)
+      _ = require(eventOpt.isDefined,
```

Comment:
It seems like all the return types should just reflect the possibility the event isn't defined with a `Future[Option[...]]` ?

Not just this method, but the other `signEvent()` methods too.

Reply:
This would be better for a future PR.

However, I disagree with doing this. I think it makes more sense to throw here.

Reply:
I disagree, the "error" is the the `eventName` doesn't exist, which is what `Option` is meant to represent (this thing _may_ exist, or it may not, in which case we give you `None`).

Reply:
Actually, I think you are right. This is signing for something they think exists in the database. Nvm.